### ISSUE TITLE
Incomplete checks for color contrast on text with ellipsis

### ISSUE THREAD

**Issue by just-boris** [dequelabs/axe-core#2669]
When text has `text-overflow: ellipsis`, axe checker fails to complete the color contrast check under some conditions:

When the truncated element is the last child, it works fine:

```html
<p class="truncated">
  This text will pass! And more text here...
</p>
```

However, if there will be an extra `span` inside, the color contrast check will fail

```html
<p class="truncated">
  <span>This text will fail! And more text here...</span>
</p>
```

Live demo: https://output.jsbin.com/comuhaq
There are two nodes, one passes the check and the other is not. The expected result is that they both will pass.
Also note the extra styles at the `body` selector. They are required to reproduce the issue.

```
axe-core version: 4.0.2
chrome extension: 4.6.1
```

**Comment by straker** [comment]
Thanks for the issue. I can see where the problem is happening but will need to investigate it further to see what we can do to resolve it.

**Comment by straker** [comment]
Note about the problem: It seems our `visuallyContains` code doesn't think that the `span` is contained within either the parent `div` or the `body` since the width of the `span` is larger than the page.
